samcli/lib/utils/stream_writer.py

```diff
@@ -17,12 +18,35 @@ def __init__(self, stream: TextIO, auto_flush: bool = False):
             Whether to autoflush the stream upon writing
         """
         self._stream = stream
+        self._stream_bytes = stream_bytes
         self._auto_flush = auto_flush
+        if isinstance(stream, TextIOWrapper):
+            self._stream_bytes = stream
 
     @property
     def stream(self) -> TextIO:
         return self._stream
 
+    def write_bytes(self, output: bytes):
+        """
+        Writes specified text to the underlying stream
+        Parameters
+        ----------
+        output bytes-like object
+            Bytes to write into buffer
+        """
+        # all these ifs are to satisfy the linting/type checking
+        if self._stream_bytes:
+            if isinstance(self._stream_bytes, TextIOWrapper):
+                self._stream_bytes.buffer.write(output)
+                if self._auto_flush:
+                    self._stream_bytes.flush()
+
+            elif isinstance(self._stream_bytes, BytesIO):
+                self._stream_bytes.write(output)
+                if self._auto_flush:
+                    self._stream_bytes.flush()
```

Comment:
Can we use early return at least from the first `if` statement?
```suggestion
        if not self._stream_bytes:
            return
        if isinstance(self._stream_bytes, TextIOWrapper):
            self._stream_bytes.buffer.write(output)
            if self._auto_flush:
                self._stream_bytes.flush()

        elif isinstance(self._stream_bytes, BytesIO):
            self._stream_bytes.write(output)
            if self._auto_flush:
                self._stream_bytes.flush()
```